compiler/src/dotty/tools/dotc/ast/TreeTypeMap.scala

```diff
@@ -68,8 +68,13 @@ class TreeTypeMap(
     }
   }
 
-  def mapType(tp: Type): Type =
-    mapOwnerThis(typeMap(tp).substSym(substFrom, substTo))
+  val mapType: Type => Type =
+    val substMap = new TypeMap():
+      def apply(tp: Type): Type = tp match
+        case tp: TermRef if tp.symbol.isImport => mapOver(tp)
+        case tp => tp.substSym(substFrom, substTo)
+    typeMap.andThen(substMap).andThen(mapOwnerThis)
```

Comment:
I believe `andThen` is not inlined. So in the interest of performance I suggest the less nice
```scala
def mapType(tp: Type): Type = 
  ...
  mapOwnerThis(substMap(typeMap(tp)))
```
We allocate fewer closures that way, and empirically that matters (a bit)

Reply:
Indeed ! turns out it's even `@annotation.unspecialized`